tests/testthat/custom_tests/regional-dataset-specific.R

```diff
@@ -25,3 +25,52 @@ test_who_level_1 <- function(region) {
 
 # Add data set specific custom tests here in the format of above using
 # variables from test_regional_dataset() environment
+test_UK_level_1 <- function(region) {
+  data_name <- "UK level 1 with 'nhsregions=TRUE'"
+  region$nhsregions <- TRUE
+  region$nhsregions <- TRUE
+  source <- class(region)[1]
+  nhs_included_path <- paste0(
+    "custom_data/", source,
+    "_level_", region$level, "_nhs", ".rds"
+  )
+  if (!file.exists(nhs_included_path)) {
+    download <- TRUE
+  } else {
+    download <- FALSE
+  }
+  if (download) {
+    test_that(paste(data_name, " downloads sucessfully"), { # nolint
+      region$download()
+      expect_s3_class(region$nhs_raw, "data.frame")
+      expect_true(nrow(region$nhs_raw) > 0)
+      expect_true(ncol(region$nhs_raw) >= 2)
+    })
+    region$nhs_raw <- dplyr::slice_tail(region$nhs_raw, n = 1000)
+    saveRDS(region$nhs_raw, nhs_included_path)
+  } else {
+    region$nhs_raw <- readRDS(nhs_included_path)
+  }
+  test_that(paste(data_name, "can be cleaned as expected"), {
+    region$clean()
+    expect_s3_class(region$data$clean, "data.frame")
+    expect_true(nrow(region$data$clean) > 0)
+    expect_true(ncol(region$data$clean) >= 2)
+    expect_clean_cols(region$data$clean, level = region$level)
```

Comment:
Good job with this. Looks good. Might need to add a custom download section to the framework to make this easier for others potentially?

Reply:
I agree, should I add one here or raise a seperate issue?

Reply:
II raised a seperate issue to handle this #202